### DISCUSSION TITLE
On Delta UV computation?

### DISCUSSION THREAD

**Discussion by blue-j** [colour-science/colour#1119] [Q&A] [answered]
Is this legit?

https://www.waveformlighting.com/tech/calculate-duv-from-cie-1931-xy-coordinates

No citation for the calculation is provided.  Does Colour-science have this?  I could not find it in the API reference.

Thanks in advance for any consideration you provide this post!

- J

**Comment by tjdcs** [comment, marked as answer]
We don't have this particular calculation, but that's ok because it's just an approximation function anyway. It's more accurate to do the correct conversions. 

Convert from xy to UCS uv, and then compute uv to CCT with the defaut options.

```python
from colour import xy_to_UCS_uv, uv_to_CCT

cct_duv = uv_to_CCT(xy_to_UCS_uv((0.3143, 0.3275)))
print(f"CCT: {cct_duv[0]:.1f}, ∆uv: {cct_duv[1]:.4f}")
```

**Reply by blue-j** [reply]
You're kind to share your knowledge, thank you!  :)   Is there any reason to calculate Duv for red primary for example?  Is this really only about white?  I can't see how someone would care about the distance to the Planckian locus for a red point, but I want to be thorough in what I provide our users.

**Reply by thomasmansencal** [reply]
You could compute it for any value but the concept of CCT is only valid within 0.05 Duv

**Reply by blue-j** [reply]
for RGB color spaces, does CCT = illuminant in K?  it seems to.

**Reply by thomasmansencal** [reply]
CCT stands for Correlated Colour Temperature, i.e. Colour Temperature related to a blackbody with an offset normal to the blackbody locus.
